scripts/imgtool/main.py

```diff
@@ -154,6 +154,23 @@ def validate_version(ctx, param, value):
         raise click.BadParameter("{}".format(e))
 
 
+def validate_security_counter(ctx, param, value):
+    if value is not None:
+        if value == '':
+            return value
+        else:
+            try:
+                if value[:2].lower() == '0x':
+                    return int(value[2:], 16)
+                elif value[:1] == '0':
+                    return int(value, 8)
+                else:
+                    return int(value, 10)
```

Comment:
This conversion code is the same as in the `convert` function inside `BasedIntParamType`. You can create a new global function `convert_based_int(value)` that does the conversion (or throws an exception) and call it here and inside `BasedIntParam:convert`.

Reply:
Please, just use `return int(value, 0)`, this functionality is already builtin to Python.